### PR TITLE
[HV] Update roadmap

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@gsmet You could add these note-worthy issues for 5.3:

* Extend programmatic API to allow alteration of ConstraintValidator list for a given constraint (HV-501)
* Support for Java 9
* Arbitrary contextual data can be added to constraint violations (HV-1020)
* Rules for parameter constraints can optionally be relaxed (HV-872)

**Comment by gunnarmorling** [comment]
And how about adding a link like this to have the list of all issues of a given past minor (5.3, 5.4. 6.0):

[List of all 5.3 issues](https://hibernate.atlassian.net/issues/?jql=project%20%3D%20HV%20AND%20fixVersion%20in%20(5.3.0.Alpha1%2C%205.3.0.CR1%2C%205.3.0.Final%2C%205.3.1.Final%2C%205.3.2.Final%2C%205.3.3.Final%2C%205.3.4.Final%2C%205.3.5.Final)%20ORDER%20BY%20issuetype%20DESC%2C%20priority%20DESC%2C%20updated%20DESC)

File: validator/roadmap.adoc [hunk 1]
```diff
@@ -7,29 +7,35 @@ Hardy Ferentschik
...
+== Hibernate Validator 6.1
+
+Engine::
+Explore taking some distance from the reflection API (e.g. dealing with `Type` is rather confusing, replace some of the types with memory-friendly counterparts (`Method`?), do not rely on `TypeVariable`). Note that it might also help with the future usage of Jandex for annotation discovery.
+
+Programmatic API::
+Be able to define constraint on properties using lambdas (ie without constraint annotations)
+
+Java 9::
+Usage of method handles for accessing bean states (https://hibernate.atlassian.net/browse/HV-1226[HV-1226])
```
Comment (gunnarmorling) [inline validator/roadmap.adoc:34]: s/method handles/var and method handles/

Also let's add

* "Explore usage of [ResourceBundleProvider](http://download.java.net/java/jdk9/docs/api/java/util/spi/ResourceBundleProvider.html)" for obtaining validation error messages from other modules
* Providing HV with JDK 9 module descriptors

File: validator/roadmap.adoc [hunk 2]
```diff
@@ -7,29 +7,35 @@ Hardy Ferentschik
...
+
+== Future versions
+
+Engine::
+Explore the ability to validate an object and a list of changes (https://hibernate.atlassian.net/browse/BVAL-214[BVAL-214]) +
+Explore the usage of Jandex to build the metadata (https://hibernate.atlassian.net/browse/HV-644[HV-644])
```
Comment (gunnarmorling) [inline validator/roadmap.adoc:40]: A link to Jandex and short braces with explanation of what it is would be nice.

File: validator/roadmap.adoc [hunk 3]
```diff
@@ -7,29 +7,35 @@ Hardy Ferentschik
...
+
 You can find a fine grained roadmap in our https://hibernate.atlassian.net/browse/HV[issue tracker],
 but this page is a good starting point to see where we are going.
 
-== Hibernate Validator 5.3
+== Hibernate Validator 6.0.x
 
-OSGi::
-Expose Validator as OSGi service (https://hibernate.atlassian.net/browse/HV-570[HV-570])
+Engine::
```
Comment (gunnarmorling) [inline validator/roadmap.adoc:17]: Performance improvements should be mentioned

File: validator/roadmap.adoc (hunk 2 above)
Comment (gunnarmorling) [reply validator/roadmap.adoc:40]: Also let's add: constraint ordering and explore support for other models than JavaBeans (HV-1000)

File: validator/roadmap.adoc (hunk 3 above)
Comment (gunnarmorling) [reply validator/roadmap.adoc:17]: And what about the pending PR about the SPI for what's a bean property?

**Review by gunnarmorling — COMMENTED** [review]
@gsmet Added some remarks.

For 6.0, it could be a very condensed form of the "What's new?" section in the spec.

**Comment by gsmet** [comment]
Included the items mentioned by Gunnar and merged.

BTW:
- we now have a link to the Releases page at the top of the page for the already released series
- in the series page, we have a link to all the issues fixes in the series (and we also have links for each micro version)

**Pull request closed without merging**